packages/@aws-cdk/assert/lib/assertions/have-resource-matchers.ts

```diff
@@ -281,6 +281,90 @@ export function notMatching(matcher: any): PropertyMatcher {
   });
 }
 
+export type TypeValidator<T> = (x: any) => x is T;
+
+/**
+ * Captures a value onto an object if it matches a given inner matcher
+ *
+ * @example
+ *
+ * const someValue = Capture.aString();
+ * expect(stack).toHaveResource({
+ *    // ...
+ *    Value: someValue.capture(stringMatching('*a*')),
+ * });
+ * console.log(someValue.capturedValue);
+ */
+export class Capture<T=any> {
```

Comment:
*sigh* one more thing to consider when rewriting this module as a JSII-able module. 

I would've asked - why not just

```ts
const notThisHash = ...;

expect(stack).not.toHaveResourceLike('AWS::CodeBuild::Project', {
  ...
  Value: notThisHash
});
```

But looking at the usage and the deep nesting, I understand why you had to do this.

Reply:
The functionality is legit and should be ported, I feel.

The jsii-able version of this might drop the type assertions but I didn't see any point in having this wobbly-typed since this module is long past jsii-ability anyway.

Reply:
Fine by me.

We'll definitely consider for this to be ported. Just wanted to make note that the form might have to change.